src/main/java/com/epam/service/TourOfferService.java

```diff
@@ -0,0 +1,63 @@
+package com.epam.service;
+
+import com.epam.model.TourOffer;
+import com.epam.repository.TourOfferDAO;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.sql.Date;
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class TourOfferService {
+
+    private final TourOfferDAO tourOfferDAO;
+
+    @Autowired
+    public TourOfferService(TourOfferDAO tourOfferDAO){
+        this.tourOfferDAO = tourOfferDAO;
+    }
+
+
+    public List<TourOffer> getTours(){
+            return tourOfferDAO.getTours();
+    }
+
+    public int deleteTour(Integer tourId){
+         return tourOfferDAO.deleteTour(tourId);
+    }
+
+    public int addTour(Integer tourId, String tourType, Date startDate, Date endDate, Integer pricePerUnit, Integer hotel_id, String description, Integer discount_id){
```

Comment:
Maybe try to use object of TourOffer instead of so many parameters in the signature?

Reply:
Got it, but who ill build this object? Is that type of task for controller layer?

Reply:
Usually this object came from client side, but in our case we can build it in controller layer

Reply:
Got it.